Jenkinsfile-upgrade

```diff
@@ -0,0 +1,138 @@
+        stage('checkout') {
+            steps {
+                checkout scm
+                sh 'rm -rf artifacts && mkdir -p artifacts'
+            }
+        }
+        stage('start openshift') {
+            steps {
+                timeout(time: 10, unit: 'MINUTES') {
+                    script {
+                        lib.startOpenshift()
+                    }
+                }
+            }
+        }
+        stage('deploy enmasse'){
+            environment {
+                JOB_NAME_SUB = "${String.format("%.15s", JOB_NAME)}"
+                OPENSHIFT_PROJECT = "${JOB_NAME_SUB}${BUILD_NUMBER}"
+                COMMIT = "${params.START_VERSION}"
+            }
+            steps{
+                withCredentials([string(credentialsId: 'openshift-host', variable: 'OPENSHIFT_URL'), usernamePassword(credentialsId: 'openshift-credentials', passwordVariable: 'OPENSHIFT_PASSWD', usernameVariable: 'OPENSHIFT_USER')]) {
+                    sh "git checkout tags/${params.START_VERSION}"
+                    sh 'make templates || true'
+                    sh "./systemtests/scripts/deploy_enmasse.sh false 'templates/build/enmasse-latest' true true"
+                }
+            }
+        }
+        stage('systemtests before upgrade') {
+            environment {
+                JOB_NAME_SUB = "${String.format("%.15s", JOB_NAME)}"
+                OPENSHIFT_PROJECT = "${JOB_NAME_SUB}${BUILD_NUMBER}"
+                USE_DUMMY_ADDRESS = true
+                SKIP_CLEANUP = true
+                SYSTEMTEST_UPGRADED = false
+            }
+            steps {
+                withCredentials([string(credentialsId: 'openshift-host', variable: 'OPENSHIFT_URL'), usernamePassword(credentialsId: 'openshift-credentials', passwordVariable: 'OPENSHIFT_PASSWD', usernameVariable: 'OPENSHIFT_USER')]) {
+                    script {
+                        lib.runSystemtests(env.CORES_DIR, 'latest', 'systemtests-upgrade', params.TEST_CASE, true)
+                    }
+                }
+            }
+        }
+        stage('upgrade enmasse'){
+            environment {
+                JOB_NAME_SUB = "${String.format("%.15s", JOB_NAME)}"
+                OPENSHIFT_PROJECT = "${JOB_NAME_SUB}${BUILD_NUMBER}"
+                DOCKER_REGISTRY = credentials('docker-registry-host')
+                DOCKER_CREDENTIALS = credentials('docker-registry-credentials')
+                DOCKER_PASS = "${env.DOCKER_CREDENTIALS_PSW}"
+                DOCKER_USER = "${env.DOCKER_CREDENTIALS_USR}"
+                COMMIT = "${params.BUILD_TAG}"
+            }
+            steps{
+                withCredentials([string(credentialsId: 'openshift-host', variable: 'OPENSHIFT_URL'), usernamePassword(credentialsId: 'openshift-credentials', passwordVariable: 'OPENSHIFT_PASSWD', usernameVariable: 'OPENSHIFT_USER')]) {
+                    script {
+                        sh "git checkout ${params.COMMIT_SHA}"
+                        sh 'make templates || true'
```

Comment:
Should probably perform a `make -C templates clean` first, to ensure that installation files are not polluted from previous version.